src/test/kotlin/org/rust/lang/core/type/RsExpressionTypeInferenceTest.kt

```diff
@@ -810,4 +810,16 @@ class RsExpressionTypeInferenceTest : RsTypificationTestBase() {
             a
         } //^ <unknown>
     """)
+
+    fun `test impl Trait`() = testExpr("""
```

Comment:
Hmm, this test is better to move to `RsTypeAwareResolveTest`, because we really want to check that `bar` is resolved to the method we want. I also suggest to add there similar test, but with multiple traits `impl Trait1+Trait2`

Reply:
Hm maybe done. For the Trait1+Trait2 case it looks like i can't have multiple `//^`in one test. So I made 2 tests testing the same source code but with different markers.